core/env/spiffe.go

```diff
@@ -22,3 +22,11 @@ func SpiffeSocketUrl() string {
 	}
 	return p
 }
+
+func SpiffeTrustDomain() string {
```

Comment:
If defined, the system will assume this one.

Of course, SPIRE should be configured accordingly for the new trust domain to work without issues.